.github/workflows/ci.yml

```
@@ -0,0 +1,123 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+  pull_request:
+
+jobs:
+  shellcheck:
+    name: Shellcheck
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run ShellCheck
+        run: |
+          sudo apt-get install shellcheck
+          shopt -s globstar; shellcheck script/**/*.sh
+
+  build:
+    name: ${{ matrix.friendlyName }} ${{ matrix.arch }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-10.15, windows-2019, ubuntu-18.04]
+        arch: [32, 64]
+        include:
+          - os: macos-10.15
+            friendlyName: macOS
+            targetPlatform: macOS
+          - os: windows-2019
+            friendlyName: Windows
+            targetPlatform: win32
+          - os: ubuntu-18.04
+            friendlyName: Linux
+            targetPlatform: ubuntu
+        exclude:
+          - os: macos-10.15
+            arch: 32
+          - os: ubuntu-18.04
+            arch: 32
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+          # Needed for script/package.sh to work
+          fetch-depth: 0
+      - name: Install dependencies
+        run: npm install
+      - name: Check formatting
+        run: npm run prettier
+      - name: Build tools
+        run: npm run check
+      - name: Install extra dependencies for building Git on Ubuntu
+        if: matrix.targetPlatform == 'ubuntu'
+        run: sudo apt-get install libcurl4-openssl-dev libexpat1-dev gettext
+      - name: Build
+        shell: bash
+        run: script/build.sh
+        env:
+          TARGET_PLATFORM: ${{ matrix.targetPlatform }}
+          TARGET_ARCH: ${{ matrix.arch }}
+      - name: Package
+        shell: bash
+        run: script/package.sh
+        env:
+          TARGET_PLATFORM: ${{ matrix.targetPlatform }}
+          TARGET_ARCH: ${{ matrix.arch }}
+      - name: Upload ouptut artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name:
+            dugite-native-${{ matrix.targetPlatform }}-${{ matrix.arch }}-output
+          path: ./output
+          retention-days: 5
+
+  release:
+    name: Create GitHub release
+    needs: build
```

Can we make the release depend on both a successful build and a valid shellcheck run?
```suggestion
    needs: [build, shellcheck]
```

---

Sounds good!